File: TrackProyectosWebAPI/DTOs/proyectoDTO.cs
using System;

namespace TrackProyectosWebAPI.DTOs
{
    public class ProyectoDTO
    {

        public int Id  { get; set; }

        public string Nombre { get; set; }

        public string Descripcion { get; set; }

        public DateTime FechaInicio { get; set; }

        public DateTime FechaFinalizacion { get; set; }
        
        public string Link  { get; set; }

        public int ProgramadorID { get; set; }
    }
}

File: TrackProyectosFront/src/app/_models/ProyectoDTO.ts
export class ProyectoDTO{
    
    id: number;
    nombre: string;
    descripcion: string;
    fechaInicio: Date;
    fechaFinalizacion: Date;
    link: string;
    programadorID: number;

constructor() {
    this.id = 0;
    this.nombre = '';
    this.descripcion = '';
    this.link = '';
    this.programadorID = 0;
}

public clone(): ProyectoDTO {
    let clonedModel: ProyectoDTO = new ProyectoDTO();
    clonedModel.id = this.id;

    return clonedModel;
  }

}